asm_to_pil/src/vm_to_constrained.rs

```diff
@@ -930,6 +934,34 @@ impl<T: FieldElement> ASMPILConverter<T> {
     }
 }
 
+/// Transforms `x` -> `x(i)` and `x' -> `x(i + 1)`
+fn transform_references_in_free_input_query<T: FieldElement>(
+    mut e: Expression<T>,
+) -> Expression<T> {
+    // TOOD we should check that we only transform columns and not other symbols.
```

Comment:
At the current point, I don't really know how to do this. The whole asm to pil conversion makes so many assumptions about which names are free to use, I think this is fine...